sdk-actors/src/main/java/io/dapr/actors/runtime/DaprStateAsyncProvider.java

```diff
@@ -56,7 +57,7 @@
   DaprStateAsyncProvider(DaprClient daprClient, DaprObjectSerializer stateSerializer) {
     this.daprClient = daprClient;
     this.stateSerializer = stateSerializer;
-    this.isStateSerializerDefault = stateSerializer.getClass() == DefaultObjectSerializer.class;
+    this.isStateSerializerJson = DefaultObjectSerializer.JSON_CONTENT_TYPE.equals(stateSerializer.getContentType());

```

Comment:
The reason why the default serializer is not recommended for production use is that it becomes a contract from the SDK. Users can bring serialize infinite different classes and our serializer cannot necessarily work for all of them. Plus, it becomes harder when it comes to persisted data, where a change in the serializer can stop applications from reading their persisted data. See discussion in https://github.com/dapr/dapr/issues/2905

When should the app implement their own serializer?
Always, unless it is just trying the Dapr Java SDK. Plus, serializers should be versioned (mark/metadata) if data is persisted, so breaking changes can be handled gracefully.

We actually considered moving the default serializer into the examples package. I am considering doing that change now to avoid confusion, which means that the default constructor for DaprBuilder would be deprecated.
I've created two issues proposing changes that should help clarify that: https://github.com/dapr/java-sdk/issues/506 https://github.com/dapr/java-sdk/issues/505

Reply:
You've just touched in a key point. Jackson is an internal implementation detail and not part of the SDK contract. Theoretically, we could replace Jackson without breaking apps.

This decision is very important because it defines where the draw the line of what to support in the future. The Dapr SDK should not be a Facade for Jackson because we would need to support it in the long term. I actually created an issue to give an example of what a "good" production-ready serializer can look like: https://github.com/dapr/java-sdk/issues/505

Reply:
Now I understand your point. You a right. If I just implement an interface DaprObjectSerializer I can use jsoniter instead of jackson and have a speed up about two times or more. Thank you. It is a good thought.